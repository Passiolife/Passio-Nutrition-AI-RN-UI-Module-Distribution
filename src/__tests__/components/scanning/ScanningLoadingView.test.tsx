// @ts-nocheck
import React from 'react';
import renderer from 'react-test-renderer';
import {
  BrandingProvider,
  QuickScanningLoadingView,
  ServicesProvider,
} from '@passiolife/nutrition-ai-ui-ux';
import { mockBranding, mockServices } from '../../provider/MockProviders';
describe('ScanningLoadingView', () => {
  it('renders correctly', async () => {
    const tree = renderer.create(
      <ServicesProvider services={mockServices}>
        <BrandingProvider branding={mockBranding}>
          <QuickScanningLoadingView />
        </BrandingProvider>
      </ServicesProvider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
