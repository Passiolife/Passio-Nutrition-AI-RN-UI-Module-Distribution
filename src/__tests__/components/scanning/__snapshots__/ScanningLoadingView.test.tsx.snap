// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanningLoadingView renders correctly 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#00000066",
      "borderRadius": 10,
      "bottom": 120,
      "flex": 1,
      "minHeight": 180,
      "padding": 5,
      "position": "absolute",
      "shadowColor": "black",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 10,
      "width": 726,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 120,
        "padding": 25,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <LottieAnimationView
        autoPlay={true}
        cacheComposition={true}
        colorFilters={[]}
        enableMergePathsAndroidForKitKatAndAbove={false}
        loop={true}
        onAnimationFailure={[Function]}
        onAnimationFinish={[Function]}
        onAnimationLoaded={[Function]}
        progress={0}
        resizeMode="contain"
        sourceJson="{"v":"5.1.20","fr":30,"ip":1,"op":122,"w":400,"h":400,"nm":"Loading","ddd":0,"assets":[],"layers":[{"ddd":0,"ind":1,"ty":4,"nm":"Shape Layer 2","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":1,"k":[{"i":{"x":[0.59],"y":[1]},"o":{"x":[0.44],"y":[0]},"n":["0p59_1_0p44_0"],"t":0,"s":[0],"e":[360]},{"i":{"x":[0.59],"y":[1]},"o":{"x":[0.41],"y":[0]},"n":["0p59_1_0p41_0"],"t":63,"s":[360],"e":[720]},{"t":123}],"ix":10},"p":{"a":0,"k":[200,200,0],"ix":2},"a":{"a":0,"k":[-3,35,0],"ix":1},"s":{"a":0,"k":[80,80,100],"ix":6}},"ao":0,"shapes":[{"ty":"gr","it":[{"d":1,"ty":"el","s":{"a":0,"k":[398,398],"ix":2},"p":{"a":0,"k":[0,0],"ix":3},"nm":"Ellipse Path 1","mn":"ADBE Vector Shape - Ellipse","hd":false},{"ty":"st","c":{"a":0,"k":[0,0.2,0.5372549019607843,1],"ix":3},"o":{"a":0,"k":100,"ix":4},"w":{"a":0,"k":50,"ix":5},"lc":2,"lj":1,"ml":4,"nm":"Stroke 1","mn":"ADBE Vector Graphic - Stroke","hd":false},{"ty":"tr","p":{"a":0,"k":[-3,35],"ix":2},"a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[100,100],"ix":3},"r":{"a":0,"k":0,"ix":6},"o":{"a":0,"k":100,"ix":7},"sk":{"a":0,"k":0,"ix":4},"sa":{"a":0,"k":0,"ix":5},"nm":"Transform"}],"nm":"Ellipse 1","np":3,"cix":2,"ix":1,"mn":"ADBE Vector Group","hd":false},{"ty":"tm","s":{"a":1,"k":[{"i":{"x":[0.59],"y":[1]},"o":{"x":[0.41],"y":[0]},"n":["0p59_1_0p41_0"],"t":63,"s":[0],"e":[100]},{"t":123}],"ix":1},"e":{"a":1,"k":[{"i":{"x":[0.56],"y":[1]},"o":{"x":[0.44],"y":[0]},"n":["0p56_1_0p44_0"],"t":0,"s":[0],"e":[100]},{"t":63}],"ix":2},"o":{"a":0,"k":1.6,"ix":3},"m":2,"ix":2,"nm":"Trim Paths 1","mn":"ADBE Vector Filter - Trim","hd":false}],"ip":-5,"op":2087,"st":-5,"bm":0},{"ddd":0,"ind":2,"ty":4,"nm":"Shape Layer 3","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":1,"k":[{"i":{"x":[0.56],"y":[1]},"o":{"x":[0.44],"y":[0]},"n":["0p56_1_0p44_0"],"t":0,"s":[0],"e":[360]},{"i":{"x":[0.59],"y":[1]},"o":{"x":[0.167],"y":[0]},"n":["0p59_1_0p167_0"],"t":61,"s":[360],"e":[360]},{"i":{"x":[0.59],"y":[1]},"o":{"x":[0.41],"y":[0]},"n":["0p59_1_0p41_0"],"t":67,"s":[360],"e":[720]},{"t":123}],"ix":10},"p":{"a":0,"k":[200,200,0],"ix":2},"a":{"a":0,"k":[-3,35,0],"ix":1},"s":{"a":0,"k":[80,80,100],"ix":6}},"ao":0,"shapes":[{"ty":"gr","it":[{"d":1,"ty":"el","s":{"a":0,"k":[398,398],"ix":2},"p":{"a":0,"k":[0,0],"ix":3},"nm":"Ellipse Path 1","mn":"ADBE Vector Shape - Ellipse","hd":false},{"ty":"st","c":{"a":0,"k":[0.1568627450980392,0.4235294117647059,0.8862745098039215,1],"ix":3},"o":{"a":0,"k":100,"ix":4},"w":{"a":0,"k":50,"ix":5},"lc":2,"lj":1,"ml":4,"nm":"Stroke 1","mn":"ADBE Vector Graphic - Stroke","hd":false},{"ty":"tr","p":{"a":0,"k":[-3,35],"ix":2},"a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[100,100],"ix":3},"r":{"a":0,"k":0,"ix":6},"o":{"a":0,"k":100,"ix":7},"sk":{"a":0,"k":0,"ix":4},"sa":{"a":0,"k":0,"ix":5},"nm":"Transform"}],"nm":"Ellipse 1","np":3,"cix":2,"ix":1,"mn":"ADBE Vector Group","hd":false},{"ty":"tm","s":{"a":0,"k":0,"ix":1},"e":{"a":1,"k":[{"i":{"x":[0.56],"y":[1]},"o":{"x":[0.44],"y":[0]},"n":["0p56_1_0p44_0"],"t":0,"s":[0],"e":[100]},{"t":61}],"ix":2},"o":{"a":0,"k":1.6,"ix":3},"m":2,"ix":2,"nm":"Trim Paths 1","mn":"ADBE Vector Filter - Trim","hd":false},{"ty":"tm","s":{"a":1,"k":[{"i":{"x":[0.59],"y":[1]},"o":{"x":[0.41],"y":[0]},"n":["0p59_1_0p41_0"],"t":67,"s":[0],"e":[100]},{"t":123}],"ix":1},"e":{"a":0,"k":100,"ix":2},"o":{"a":0,"k":0,"ix":3},"m":1,"ix":3,"nm":"Trim Paths 2","mn":"ADBE Vector Filter - Trim","hd":false}],"ip":-2,"op":2094,"st":2,"bm":0},{"ddd":0,"ind":3,"ty":4,"nm":"Shape Layer 1","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":1,"k":[{"i":{"x":[0.56],"y":[1]},"o":{"x":[0.44],"y":[0]},"n":["0p56_1_0p44_0"],"t":0,"s":[0],"e":[360]},{"i":{"x":[0.59],"y":[1]},"o":{"x":[0.167],"y":[0]},"n":["0p59_1_0p167_0"],"t":59,"s":[360],"e":[360]},{"i":{"x":[0.59],"y":[1]},"o":{"x":[0.41],"y":[0]},"n":["0p59_1_0p41_0"],"t":71,"s":[360],"e":[720]},{"t":123}],"ix":10},"p":{"a":0,"k":[200,200,0],"ix":2},"a":{"a":0,"k":[-3,35,0],"ix":1},"s":{"a":0,"k":[80,80,100],"ix":6}},"ao":0,"shapes":[{"ty":"gr","it":[{"d":1,"ty":"el","s":{"a":0,"k":[398,398],"ix":2},"p":{"a":0,"k":[0,0],"ix":3},"nm":"Ellipse Path 1","mn":"ADBE Vector Shape - Ellipse","hd":false},{"ty":"st","c":{"a":0,"k":[0.29411764705882354,0.7137254901960784,1,1],"ix":3},"o":{"a":0,"k":100,"ix":4},"w":{"a":0,"k":50,"ix":5},"lc":2,"lj":1,"ml":4,"nm":"Stroke 1","mn":"ADBE Vector Graphic - Stroke","hd":false},{"ty":"tr","p":{"a":0,"k":[-3,35],"ix":2},"a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[100,100],"ix":3},"r":{"a":0,"k":0,"ix":6},"o":{"a":0,"k":100,"ix":7},"sk":{"a":0,"k":0,"ix":4},"sa":{"a":0,"k":0,"ix":5},"nm":"Transform"}],"nm":"Ellipse 1","np":3,"cix":2,"ix":1,"mn":"ADBE Vector Group","hd":false},{"ty":"tm","s":{"a":0,"k":0,"ix":1},"e":{"a":1,"k":[{"i":{"x":[0.56],"y":[1]},"o":{"x":[0.44],"y":[0]},"n":["0p56_1_0p44_0"],"t":0,"s":[0],"e":[100]},{"t":59}],"ix":2},"o":{"a":0,"k":1.6,"ix":3},"m":2,"ix":2,"nm":"Trim Paths 1","mn":"ADBE Vector Filter - Trim","hd":false},{"ty":"tm","s":{"a":1,"k":[{"i":{"x":[0.59],"y":[1]},"o":{"x":[0.41],"y":[0]},"n":["0p59_1_0p41_0"],"t":71,"s":[0],"e":[100]},{"t":123}],"ix":1},"e":{"a":0,"k":100,"ix":2},"o":{"a":0,"k":0,"ix":3},"m":1,"ix":3,"nm":"Trim Paths 2","mn":"ADBE Vector Filter - Trim","hd":false}],"ip":0,"op":2092,"st":0,"bm":0},{"ddd":0,"ind":4,"ty":1,"nm":"White Solid 2","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[200,200,0],"ix":2},"a":{"a":0,"k":[960,540,0],"ix":1},"s":{"a":0,"k":[100,100,100],"ix":6}},"ao":0,"sw":1920,"sh":1080,"sc":"#ffffff","ip":2092,"op":2092,"st":0,"bm":0,"hidden":0}],"markers":[]}"
        speed={1}
        style={
          [
            {
              "height": 50,
              "width": 50,
            },
            undefined,
          ]
        }
        textFiltersAndroid={[]}
        textFiltersIOS={[]}
        useNativeLooping={false}
      />
    </View>
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "#4A4A4A",
          "fontSize": 18,
          "fontWeight": "600",
          "paddingHorizontal": 16,
        }
      }
    >
      Scanning...
    </Text>
  </View>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#FFFFFF",
        "fontSize": 13,
        "fontWeight": "400",
        "marginVertical": 16,
        "paddingHorizontal": 24,
        "textAlign": "center",
      }
    }
  >
    You can scan food, barcodes, and nutrition labels.
  </Text>
</View>
`;
